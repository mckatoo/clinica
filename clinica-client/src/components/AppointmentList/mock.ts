export default [
  {
    id: '1',
    photo: 'jao.jpg',
    dayOfWeek: 'terça',
    dayOfMonth: 1,
    start: '08:00',
    ending: '09:00',
    patient: 'Jão',
    attendant: 'Fernanda',
    title: 'Consulta fisioterapia'
  },
  {
    id: '2',
    photo: 'ze.jpg',
    dayOfWeek: 'quarta',
    dayOfMonth: 2,
    start: '10:00',
    ending: '11:00',
    patient: 'Zé',
    attendant: 'Fernanda',
    title: 'Seção choquinho'
  },
  {
    id: '3',
    photo: 'paulo.jpg',
    dayOfWeek: 'quarta',
    dayOfMonth: 2,
    start: '15:30',
    ending: '16:30',
    patient: 'Paulo',
    attendant: 'Maria',
    title: 'Seção choquinho'
  },
  {
    id: '4',
    photo: 'paulo.jpg',
    dayOfWeek: 'sexta',
    dayOfMonth: 4,
    start: '15:30',
    ending: '16:30',
    patient: 'Paulo',
    attendant: 'Claudia',
    title: 'Seção choquinho'
  }
]
