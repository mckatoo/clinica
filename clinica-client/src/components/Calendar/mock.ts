export default [
  {
    weekday: 'seg.',
    date: '31',
    month: '5',
    year: '2021',
    totalAppointments: 3
  },
  {
    weekday: 'ter.',
    date: '8',
    month: '6',
    year: '2021',
    totalAppointments: 2
  },
  {
    weekday: 'qui.',
    date: '10',
    month: '6',
    year: '2021',
    totalAppointments: 4
  },
  {
    weekday: 'sex.',
    date: '18',
    month: '6',
    year: '2021',
    totalAppointments: 1
  },
  {
    weekday: 'seg.',
    date: '21',
    month: '6',
    year: '2021',
    totalAppointments: 9
  },
  {
    weekday: 'qui.',
    date: '1',
    month: '7',
    year: '2021',
    totalAppointments: 10
  },
  {
    weekday: 'sex.',
    date: '2',
    month: '7',
    year: '2021',
    totalAppointments: 6
  },
  {
    weekday: 's√°b.',
    date: '3',
    month: '7',
    year: '2021',
    totalAppointments: 32
  },
]
