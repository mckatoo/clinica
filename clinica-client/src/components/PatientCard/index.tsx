import AttendingButton from 'components/AttendingButton'
import Ribbon, { RibbonColors, RibbonSizes } from 'components/Ribbon'
import Image from 'next/image'
import Link from 'next/link'

import * as S from './styles'

export type PatientCardProps = {
  id: string
  name: string
  phone: string
  address: string
  cpf: string
  rg: string
  slug: string
  birth_date: string
  photo: string
  ribbon?: React.ReactNode
  ribbonColor?: RibbonColors
  ribbonSize?: RibbonSizes
}

const PatientCard = ({
  id,
  name,
  phone,
  slug,
  photo,
  ribbon,
  ribbonColor = 'primary',
  ribbonSize = 'small'
}: PatientCardProps) => (
  <S.Wrapper>
    {!!ribbon && (
      <Ribbon color={ribbonColor} size={ribbonSize}>
        {ribbon}
      </Ribbon>
    )}
    <Link href={`/patient/${slug}`} passHref>
      <S.ImageBox>
        <Image src={photo} alt={name} layout="fill" objectFit="cover" />
      </S.ImageBox>
    </Link>
    <S.Content>
      <Link href={`/patient/${slug}`} passHref>
        <S.Info>
          <S.Title>{name}</S.Title>
          <S.Phone>{phone}</S.Phone>
        </S.Info>
      </Link>
      <S.BuyBox>
        <AttendingButton aria-label="Add to Attending List" id={id} />
      </S.BuyBox>
    </S.Content>
  </S.Wrapper>
)

export default PatientCard
