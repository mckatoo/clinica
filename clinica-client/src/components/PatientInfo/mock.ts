export default {
  id: '1',
  name: 'Milton Carlos Katoo',
  address: 'Rua Dr. Miguel Couto, 81\nVila Boa Esperan√ßa\nItapira-SP',
  phone: '(19) 99906 5094',
  photo: '../img/Milton.jpg',
  cpf: '290.325.438-90',
  rg: '30.896.256-4',
  birth_date: '30/12/1976'
}
