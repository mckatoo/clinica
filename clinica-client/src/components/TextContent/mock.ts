import { TextContentProps } from '.'

const TextContentMock: TextContentProps = {
  title: 'Address',
  content: `Rua Dr. Miguel Couto, 81\nVila Boa Esperan√ßa\nItapira-SP`
}

export default TextContentMock
