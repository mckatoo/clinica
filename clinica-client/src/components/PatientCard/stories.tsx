import { Story, Meta } from '@storybook/react/types-6-0'
// import { AttendingContextData } from 'hooks/use-attending'

import PatientCard, { PatientCardProps } from '.'

export default {
  title: 'PatientCard',
  component: PatientCard,
  parameters: {
    backgrounds: {
      default: 'ikatoo-dark'
    }
  },
  args: {
    slug: '29032543890',
    name: 'Milton Carlos Katoo',
    phone: '19-99906-5094',
    photo: 'https://source.unsplash.com/user/willianjusten/354x472'
  },
  argTypes: {
    onFav: { action: 'clicked' },
    ribbon: { type: 'string' }
  }
} as Meta

export const Default: Story<PatientCardProps> = (args) => (
  <div style={{ width: '30rem' }}>
    <PatientCard {...args} />
  </div>
)

// export const IsInAttending: Story<PatientCardProps & AttendingContextData> = (args) => (
export const IsInAttending: Story<PatientCardProps> = (args) => (
  <div style={{ width: '30rem' }}>
    <PatientCard {...args} />
  </div>
)

// IsInAttending.args = {
//   isInAttending: () => true
// }

export const WithRibbon: Story<PatientCardProps> = (args) => (
  <div style={{ width: '30rem' }}>
    <PatientCard {...args} />
  </div>
)

WithRibbon.args = {
  ribbon: '20% OFF',
  ribbonSize: 'small',
  ribbonColor: 'primary'
}
