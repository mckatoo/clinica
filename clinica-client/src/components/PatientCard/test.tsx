import 'session.mock'

import { render, screen } from 'utils/test-utils'

import PatientCard from './'

const props = {
  id: '1',
  slug: '29032543890',
  name: 'Milton Carlos Katoo',
  phone: '19-99906-5094',
  address: 'Rua Dr Miguel Couto, 81 /nVila Boa Esperan√ßa /nItapira',
  cpf: '38457638908',
  rg: '2099879087',
  birth_date: '12/12/1969',
  photo: 'https://source.unsplash.com/user/willianjusten/354x472'
}

describe('<PatientCard />', () => {
  it('should render correctly', () => {
    const { container } = render(<PatientCard {...props} />)

    expect(
      screen.getByRole('heading', { name: props.name })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: props.phone })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: props.name })).toHaveAttribute(
      'src',
      props.photo
    )

    expect(screen.getByRole('link', { name: props.name })).toHaveAttribute(
      'href',
      `/patient/${props.slug}`
    )

    expect(screen.getByLabelText(/Add to Attending/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render Ribbon', () => {
    render(
      <PatientCard
        {...props}
        ribbon='My Ribbon'
        ribbonColor='secondary'
        ribbonSize='small'
      />
    )
    const ribbon = screen.getByText(/my ribbon/i)

    expect(ribbon).toHaveStyle({ backgroundColor: '#F98A05' })
    expect(ribbon).toHaveStyle({ height: '2.6rem', fontSize: '1.2rem' })
    expect(ribbon).toBeInTheDocument()
  })
})
