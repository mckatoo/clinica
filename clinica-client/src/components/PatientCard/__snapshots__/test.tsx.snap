// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PatientCard /> should render correctly 1`] = `
.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #EBFFFD;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 3rem;
  font-size: 1.2rem;
}

.c7:focus {
  outline: 1px dashed;
}

.c7:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c7 svg {
  width: 1.5rem;
}

.c7 svg + span {
  margin-left: 0.8rem;
}

.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #EBFFFD;
}

.c1 {
  position: relative;
  min-height: 14rem;
  width: 100%;
  background: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c1 img {
  max-height: 14rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  height: 100%;
  margin: 1.6rem;
}

.c3 {
  max-width: calc(100% - 2.5rem);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-weight: 600;
  color: #030517;
}

.c5 {
  font-size: 1.4rem;
  font-weight: 600;
  color: #8F8F8F;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

<article
  class="c0"
>
  <a
    class="c1"
    href="/patient/29032543890"
  >
    <img
      alt="Milton Carlos Katoo"
      layout="fill"
      src="https://source.unsplash.com/user/willianjusten/354x472"
    />
  </a>
  <div
    class="c2"
  >
    <a
      class="c3"
      href="/patient/29032543890"
    >
      <h3
        class="c4"
      >
        Milton Carlos Katoo
      </h3>
      <h4
        class="c5"
      >
        19-99906-5094
      </h4>
    </a>
    <div
      class="c6"
    >
      <button
        aria-label="Add to attending"
        class="c7"
      >
        <svg
          aria-hidden="true"
          class="c8"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 22h14c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2h-2a1 1 0 00-1-1H8a1 1 0 00-1 1H5c-1.103 0-2 .897-2 2v15c0 1.103.897 2 2 2zM5 5h2v2h10V5h2v15H5V5z"
          />
          <path
            d="M11 13.586l-1.793-1.793-1.414 1.414L11 16.414l5.207-5.207-1.414-1.414z"
          />
        </svg>
        <span>
          Add to attending
        </span>
      </button>
    </div>
  </div>
</article>
`;
